app/grandchallenge/algorithms/models.py

```diff
@@ -414,6 +424,19 @@ def add_user(self, user):
     def remove_user(self, user):
         return user.groups.remove(self.users_group)
 
+    @property
+    def file_upload_field_url_path(self):
+        return "algorithms:file-upload"
+
+    @property
+    def civ_set_lookup(self):
+        return "inputs"
+
+    @property
+    def civ_sets_related_manager(self):
+        # should return all jobs for this algorithm (not just for the active image)
```

Comment:
Because we use this to determine whether there are existing files for a user to choose from, and for that we should take into account files linked to all past jobs of an algorithm, not just jobs linked to the current active image.

Reply:
But then why limit it just to the one algorithm?

Reply:
Good point. We can return all jobs the user has `view_job` permission for. Or should it be locked down to `change_job` permission. Difference would be that the latter would only allow re-using files they uploaded themselves (as creator of a job), where the former would allow a challenge participant to use files from a challenge phase archive if the phases is an open logs phase.  I think `view_job` permission is fine.